include/irr/asset/IImageFilter.h

```diff
@@ -16,20 +16,76 @@ namespace irr
 namespace asset
 {
 
-// runtime polymorphic
+//! Base class for general filters with runtime polymorphism.
+/*
+	Filters can execute various actions basing on input image
+	to get an output image.
+	Available filters are as following:
+
+	- Fill Filter
+	- Copy Filter
+	- Flatten Filter
+	- Convert Filter
+	- Swizzle && Convert Filter
+	- Utility Filter
+	- Blit Filter
+	- Generate Mip Maps Filter
+
```

Comment:
if you don't know what filter you'll be executing at runtime, you can use the polymorphic interface and operate on `IImageFilter` pointers or references

Reply:
adjusted in following commit